index.rst

```diff
@@ -1,61 +1,546 @@
+
+.. figure:: /_static/ConfigCSCStart.png
+   :name: fig-csc-start
+   :target: ../_images/ConfigCSCStart.png
+   :alt: Configurable CSC start process
+
+   Configurable CSC start process.
+
+settingVersions
+^^^^^^^^^^^^^^^
+
+The ``settingVersions`` event is a Generic event that is to be implemented by
+every CSC. It contains three parameters: ``recommendedSettingsLabels``,
+``recommendedSettingsVersion`` and ``settingsUrl``. This event must be published by
+all CSCs upon entering the ``STANDBY`` state and broadcast information about
+"recommended" settings (e.g. configurations). The combination of
+``recommendedSettingsLabels`` and ``recommendedSettingsVersion`` allow users to
+uniquely identify a set of configurations. A name/version setting must always
+result in the same configuration parameters. Any changes in the configuration
+repository or database should produce a new version.
+
+The ``settingsUrl`` attribute is a URL indicating how the CSC connects to its
+settings.  It will start with "file:" if it is a clone of a git repo, or the
+standard URL, if a database.
+
+The ``recommendedSettingsLabels`` will contain a comma separated list of labels,
+each label maps to a configuration. The same label can point to different
+name/version pair over time.  This information should be available in the CSC
+configuration repository or database and must match the value in
+``recommendedSettingsLabels`` published to SAL. Labels must be human readable
+strings that clearly state the purpose of that configuration (e.g. current,
+nighttime, daytime). Labels should avoid having version numbers or dates in
+them. They are group classifiers and have some relative permanence. Transient
+labels with Jira ticket numbers may be used for developing new configurations.
+They should be moved to standard type labels at the earliest opportunity.
+**The order of the labels is important**, as the first label in the list is
+assumed to be the default label for any configurable CSC.
+
+The ``recommendedSettingsVersion`` will be filled with the version information
+about the configuration repository or database. For configurations stored in
+git repositories we propose to use the command;
+
+.. prompt:: bash
+
+    git describe --all --long --always --dirty --broken
+
+as a descriptor. Versions may take any form necessary to convey the
+appropriate information. They are individual identifiers and can change
+rapidly.
+
+The configuration repository or database may contain any number of different
+configurations with different labels. Configurable CSCs must specify a list of
+recommended labels. How they implement this is up to the CSC. It should be
+noted that *not all configuration labels need to be related with a recommended
+configuration*. For instance, old configurations that are still valid can be
+kept with the repository without a label. This will allow knowledgeable users
+to use them if needed.
```

Comment:
you mean when a user uses the name of the file instead of the label? Specially in the case where the file is not linked to any label? 

You should always be able to go back to the configuration used knowing the label/configuration mapping or the name of the configuration plus the git repo descriptor. Yes, the descriptor will be changing over time  but knowing which one was used you should be able to go back to it.